frontend/src/hooks/useScript.ts

```diff
@@ -0,0 +1,83 @@
+/* Copyright 2024 Marimo. All rights reserved. */
+import React from "react";
+
+type ScriptStatus = "loading" | "ready" | "error" | "unknown";
+interface ScriptOptions {
+  removeOnUnmount?: boolean;
+}
+
+const defaultOptions = { removeOnUnmount: false };
+
+export function useScript(
+  src: string,
+  options: ScriptOptions = defaultOptions,
+) {
+  const [status, setStatus] = React.useState<ScriptStatus>();
```

Comment:
in order to avoid an extra render, can you actually check the status from the start
```ts
useState<ScriptStatus>(() => {
    return document.querySelector(`script[src="${src}"]`)?.dataset.status ?? 'loading'l
})
```